Sami/Parser/ClassVisitor/PropertyClassVisitor.php

```diff
@@ -50,58 +49,26 @@ public function visit(ClassReflection $class)
      * Adds a new property to the class using an array of tokens.
      *
      * @param ClassReflection $class       Class reflection
-     * @param string          $propertyTag Property tag contents
+     * @param string|array    $propertyTag Property tag contents
      *
      * @return bool
      */
     protected function injectProperty(ClassReflection $class, $propertyTag)
     {
-        if (!$data = $this->parseProperty($propertyTag)) {
-            return false;
-        }
-
-        $property = new PropertyReflection($data['name'], $class->getLine());
-        $property->setDocComment($data['description']);
-        $property->setShortDesc($data['description']);
-
-        if (isset($data['hint'])) {
-            $property->setHint(array(array($data['hint'], null)));
-        }
-
-        $class->addProperty($property);
-
-        return true;
-    }
-
-    /**
-     * Parses the parts of an @property tag into an associative array.
-     *
-     * @param string $tag Property tag contents
-     *
-     * @return array
-     */
-    protected function parseProperty($tag)
-    {
-        // Account for default array syntax
-        $tag = str_replace('array()', 'array', $tag);
-
-        /** @var PropertyTag $propertyTag */
-        $propertyTag = $this->context->getDocBlockParser()->getTag('@property '.$tag);
-        $propertyName = $propertyTag->getVariableName();
+        if (is_array($propertyTag) && count($propertyTag) == 3 && !empty($propertyTag[1])) {
+            $property = new PropertyReflection($propertyTag[1], $class->getLine());
+            $property->setDocComment($propertyTag[2]);
+            $property->setShortDesc($propertyTag[2]);
```

Comment:
I think you missed the fact that this is not inside `parseProperty()` anymore, since that method no longer exists because of the following reason:

Before my changes, `@property` tags were not specifically mentioned in the `switch-case` statements inside `\Sami\Parser\DocBlockParser::parseTag()`, resulting in them always hitting the `default` branch of that `switch-case`, where you just get the whole tag content as one string.  Thus the need to parse that line again.

After my changes, `@property` tags come properly parsed out of `DocBlockParser::getTag()`, so there is no need to parse them again inside the Visitor. It is also safe to assume that the array at this point is indeed a parsed `@property` tag, since it is fetched inside the `visit()` method via `ClassReflection::getTags('property')` and then passed to `injectProperty()`.

I'm pretty sure that this is correct behavior and I tested it with various projects. But again, you know the codebase of this project far better than I do, so I could of course be mislead here.